### PR TITLE
Issue/4413 onboarding generic error

### PR DESCRIPTION
**Pull request by nbradbury**
Parent issue #4413 - this PR implements the generic error state for the onboarding flow. Note that there were no designs for this screen, so a design review is required.

generic

PS: The above screenshot was updated to reflect the layout changes in 9f072b4.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/onboarding/CardReaderOnboardingViewModel.kt [hunk 1]
```diff
@@ -81,7 +83,11 @@ class CardReaderOnboardingViewModel @Inject constructor(
...
                 CardReaderOnboardingState.GenericError ->
-                    viewState.value = OnboardingViewState.GenericErrorState
+                    viewState.value = OnboardingViewState.GenericErrorState(
+                        onContactSupportActionClicked = ::onContactSupportClicked,
+                        onLearnMoreActionClicked = ::onLearnMoreClicked,
+                        onButtonActionClicked = ::exitFlow
+                    )
                 CardReaderOnboardingState.NoConnectionError ->
                     viewState.value = OnboardingViewState.NoConnectionErrorState
             }
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/onboarding/CardReaderOnboardingViewModel.kt]: Not your change, but just to consider in the next PRs - make it `.exhaustive`  maybe?

File: WooCommerce/src/main/res/layout/fragment_card_reader_onboarding_stripe.xml [hunk 2]
```diff
@@ -1,81 +1,86 @@
...
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:layout_marginTop="@dimen/major_350"
-            app:layout_constraintBottom_toBottomOf="parent"
-            app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toTopOf="parent"
-            app:layout_constraintVertical_bias="1" />
+    <include
+        android:id="@+id/learn_more_container"
```
Comment (kidinov) [inline WooCommerce/src/main/res/layout/fragment_card_reader_onboarding_stripe.xml]: nit: shell we call it `learnMoreContainer` to follow the view's IDs convention?

**Review by kidinov — APPROVED** [review]
Looks good! Just 2 things to consider:

* Shell be the content be centered between the top and "close" button? Maybe I'm wrong but it looks like it's the case for other error screens
* Maybe as part of another PR probably would be nice to check if all View IDs and the layouts (currently they are `fragment...` instead of `..._fragment` for instance) are following the convention we voted a couple of weeks ago - https://github.com/woocommerce/woocommerce-android/blob/develop/CONVENTION.md#view-ids

**Not merging because design review is needed**

**Comment by nbradbury** [comment]
> Shell be the content be centered between the top and "close" button? Maybe I'm wrong but it looks like it's the case for other error screens

@kidinov You're right, that screen is a little off. I updated it to more closely match the other onboarding screens in 9f072b4. I also made the `when` clause `exhaustive` and renamed the view to `learnMoreContainer`, as you suggested. Ready for another round!

generic

**Comment by kidinov** [comment]
@nbradbury it's approved, from my side, let's see what @woocommerce/mobile-designers will say =)

**Comment by ghost** [comment]
Hey @nbradbury @kidinov , I've done a design for Android and iOS for this error. Could we please use this instead.

CC: @Garance91540 

Screen Shot 2021-08-04 at 4 49 47 pm

**Review by Garance91540 — APPROVED** [review]
Agreeing with Adam's changes ;)

**Review by Garance91540 — CHANGES_REQUESTED** [review]
^

**Comment by nbradbury** [comment]
@adamzelinski @Garance91540 I'll update this to match the design (thank you!), but I'm wondering why we need to say "Unable to verify In-Person Payments for this store" twice?

**Comment by nbradbury** [comment]
@kidinov I've updated the screen to match the above design. If it's okay with you, I'd prefer to merge this now and if there's any feedback about the wording we can address that later.

generic

**Comment by kidinov** [comment]
@nbradbury totally ok with me =)

**Review by Garance91540 — APPROVED** [review]

**Pull request merged**